--- a/editors/cmp.c
+++ b/editors/cmp.c
@@ -12,6 +12,11 @@
 //config:	help
 //config:	cmp is used to compare two files and returns the result
 //config:	to standard output.
+//config:
+//config:config FEATURE_CMP_LONG_OPTIONS
+//config:	bool "Enable long options"
+//config:	default n
+//config:	depends on CMP && LONG_OPTS
 
 //applet:IF_CMP(APPLET(cmp, BB_DIR_USR_BIN, BB_SUID_DROP))
 
@@ -41,6 +46,15 @@
 #define CMP_OPT_l (1<<1)
 #define CMP_OPT_n (1<<2)
 
+#if ENABLE_FEATURE_CMP_LONG_OPTIONS
+static const char *cmp_long_opts =
+	"quiet\0"   No_argument       "s"
+	"silent\0"  No_argument       "s"
+	"verbose\0" No_argument       "l"
+	"bytes\0"   Required_argument "n"
+	;
+#endif
+
 int cmp_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int cmp_main(int argc UNUSED_PARAM, char **argv)
 {
@@ -54,6 +68,17 @@
 	int retval = 0;
 	int max_count = -1;
 
+#if ENABLE_FEATURE_CMP_LONG_OPTIONS
+	opt = getopt32long(argv, "^"
+			OPT_STR
+			"\0" "-1"
+			IF_DESKTOP(":?4")
+			IF_NOT_DESKTOP(":?2")
+			":l--s:s--l",
+			cmp_long_opts,
+			&max_count
+	);
+#else
 	opt = getopt32(argv, "^"
 			OPT_STR
 			"\0" "-1"
@@ -62,6 +87,7 @@
 			":l--s:s--l",
 			&max_count
 	);
+#endif
 	argv += optind;
 
 	filename1 = *argv;
