diff --git a/coreutils/stat.c b/coreutils/stat.c
index 2c2909e..6a9f9fc 100644
--- a/coreutils/stat.c
+++ b/coreutils/stat.c
@@ -18,6 +18,11 @@
 //config:	help
 //config:	display file or filesystem status.
 //config:
+//config:config FEATURE_STAT_LONG_OPTIONS
+//config:	bool "Enable long options"
+//config:	default n
+//config:	depends on STAT && LONG_OPTS
+//config:
 //config:config FEATURE_STAT_FORMAT
 //config:	bool "Enable custom formats (-c)"
 //config:	default y
@@ -115,6 +120,25 @@ enum {
 	OPT_SELINUX     = (1 << (2+ENABLE_FEATURE_STAT_FILESYSTEM)) * ENABLE_SELINUX,
 };
 
+#if ENABLE_FEATURE_STAT_LONG_OPTIONS
+static const char *stat_long_opts =
+	"deference\0"             No_argument       "L"
+	IF_FEATURE_STAT_FILESYSTEM(
+	"file-system\0"           No_argument       "f"
+	)
+	"terse\0"                 No_argument       "t"
+	"format\0"                Required_argument "c"
+	;
+#endif
+
+static const char stat_opt_str[] ALIGN1 = "^"
+	"tL"
+	IF_FEATURE_STAT_FILESYSTEM("f")
+	IF_SELINUX("Z")
+	IF_FEATURE_STAT_FORMAT("c:")
+	"\0" "-1" /* min one arg */
+;
+
 #if ENABLE_FEATURE_STAT_FORMAT
 typedef bool (*statfunc_ptr)(const char *, const char *);
 #else
@@ -768,14 +792,13 @@ int stat_main(int argc UNUSED_PARAM, char **argv)
 
 	opts =
 #endif
-	getopt32(argv, "^"
-		"tL"
-		IF_FEATURE_STAT_FILESYSTEM("f")
-		IF_SELINUX("Z")
-		IF_FEATURE_STAT_FORMAT("c:")
-		"\0" "-1" /* min one arg */
+#if ENABLE_FEATURE_STAT_LONG_OPTIONS
+	getopt32long(argv, stat_opt_str, stat_long_opts
 		IF_FEATURE_STAT_FORMAT(,&format)
 	);
+#else
+	getopt32(argv, stat_opt_str IF_FEATURE_STAT_FORMAT(,&format));
+#endif
 #if ENABLE_FEATURE_STAT_FILESYSTEM
 	if (opts & OPT_FILESYS) /* -f */
 		statfunc = do_statfs;
