--- busybox.orig/coreutils/readlink.c
+++ busybox/coreutils/readlink.c
@@ -25,14 +25,16 @@
 //kbuild:lib-$(CONFIG_READLINK) += readlink.o
 
 //usage:#define readlink_trivial_usage
-//usage:	IF_FEATURE_READLINK_FOLLOW("[-fnv] ")
+//usage:	IF_FEATURE_READLINK_FOLLOW("[-efmnv] ")
 //usage:	IF_NOT_FEATURE_READLINK_FOLLOW("[-n] ")
 //usage:	"FILE"
 //usage:#define readlink_full_usage "\n\n"
 //usage:       "Display the value of a symlink" "\n"
 //usage:     "\n	-n	Don't add newline"
 //usage:	IF_FEATURE_READLINK_FOLLOW(
+//usage:     "\n	-e	Canonicalize by following all symlinks (all components must exist)"
 //usage:     "\n	-f	Canonicalize by following all symlinks"
+//usage:     "\n	-m	Canonicalize by following all symlinks (allowing missing components)"
 //usage:     "\n	-v	Verbose"
 //usage:	)
 
@@ -64,6 +66,38 @@
  * behavior which means we'd have to implement the full stack ourselves :(.
  */
 
+#if ENABLE_FEATURE_READLINK_FOLLOW
+static char* readlink_m(const char *path)
+{
+	char *buf, *tail = NULL;
+
+	errno = 0;
+	buf = xmalloc_realpath(path);
+	while (!buf && errno == ENOENT) {
+		char *new_tail = strrchr(path, '/');
+		if (tail) {
+			*tail = '/';
+		}
+		if (new_tail) {
+			*new_tail = '\0';
+			tail = new_tail;
+			buf = xmalloc_realpath(path);
+		} else {
+			break;
+		}
+	}
+
+	if (buf && !errno && tail) {
+		unsigned len = strlen(buf);
+		buf = xrealloc(buf, len + strlen(tail + 1) + 2);
+		buf[len++] = '/';
+		strcpy(buf + len, tail + 1);
+	}
+
+	return buf;
+}
+#endif
+
 int readlink_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int readlink_main(int argc UNUSED_PARAM, char **argv)
 {
@@ -71,8 +105,8 @@ int readlink_main(int argc UNUSED_PARAM,
 	unsigned opt;
 
 	/* -n must use bit 0 (see printf below) */
-	opt = getopt32(argv, "^" "n" IF_FEATURE_READLINK_FOLLOW("fvsq")
-                       "\0" "=1");
+	opt = getopt32(argv, "^" "n" IF_FEATURE_READLINK_FOLLOW("fvsqem")
+                       "\0" "=1:e--fm:f--em:m--ef");
 
 	/* compat: coreutils readlink reports errors silently via exit code */
 	if (!(opt & 4)) /* not -v */
@@ -81,6 +115,10 @@ int readlink_main(int argc UNUSED_PARAM,
 	/* NOFORK: only one alloc is allowed; must free */
 	if (opt & 2) { /* -f */
 		buf = xmalloc_realpath_coreutils(argv[optind]);
+	} else if (opt & 32) { /* -e */
+		buf = xmalloc_realpath(argv[optind]);
+	} else if (opt & 64) { /* -m */
+		buf = readlink_m(argv[optind]);
 	} else {
 		buf = xmalloc_readlink_or_warn(argv[optind]);
 	}
